packages/swapper/src/swappercli.ts

```diff
@@ -184,55 +185,58 @@ const main = async (): Promise<void> => {
-      sellAsset,
-      buyAsset,
-      sellAmountBeforeFeesCryptoBaseUnit,
-      sendMax: false,
-      accountType: utxoAccountType || bitcoin.ChainAdapter.defaultUtxoAccountType,
-      accountNumber,
-      xpub: publicKey?.xpub || '',
-      receiveAddress: buyAssetReceiveAddr,
-    })
-  } catch (e) {
-    console.error(e)
-    return
-  }
-
-  if (!quote) {
-    console.warn('no quote returned')
-    return
-  }
-
-  const buyAmountCryptoPrecision = fromBaseUnit(
-    quote.buyAmountCryptoBaseUnit || '0',
-    buyAsset.precision,
-  )
-  const answer = readline.question(
-    `Swap ${sellAmountBeforeFeesCryptoPrecision} ${
-      sellAsset.symbol
-    } for ${buyAmountCryptoPrecision} ${buyAsset.symbol} on ${swapper.getType()}? (y/n): `,
-  )
-  if (answer === 'y') {
-    const trade = await swapper.buildTrade({
-      chainId: sellAsset.chainId as UtxoChainId,
-      wallet,
-      buyAsset,
-      sendMax: false,
-      sellAmountBeforeFeesCryptoBaseUnit,
-      sellAsset,
-      receiveAddress: buyAssetReceiveAddr,
-      accountType: utxoAccountType || bitcoin.ChainAdapter.defaultUtxoAccountType,
-      accountNumber,
-      xpub: publicKey?.xpub || '',
-    })
-
-    const tradeResult = await swapper.executeTrade({ trade, wallet })
-    console.info('broadcast tx with id: ', tradeResult.tradeId)
-  }
+  const quote = await swapper.getTradeQuote({
+    chainId: sellAsset.chainId as UtxoChainId,
+    sellAsset,
+    buyAsset,
+    sellAmountBeforeFeesCryptoBaseUnit,
+    sendMax: false,
+    accountType: utxoAccountType || bitcoin.ChainAdapter.defaultUtxoAccountType,
+    accountNumber,
+    xpub: publicKey?.xpub || '',
+    receiveAddress: buyAssetReceiveAddr,
+  })
+  return quote.match({
+    ok: async quote => {
+      const buyAmountCryptoPrecision = fromBaseUnit(
+        quote.buyAmountCryptoBaseUnit || '0',
+        buyAsset.precision,
+      )
+      const answer = readline.question(
+        `Swap ${sellAmountBeforeFeesCryptoPrecision} ${
+          sellAsset.symbol
+        } for ${buyAmountCryptoPrecision} ${buyAsset.symbol} on ${swapper.getType()}? (y/n): `,
+      )
+      if (answer === 'y') {
+        const maybeTrade = await swapper.buildTrade({
+          chainId: sellAsset.chainId as UtxoChainId,
+          wallet,
+          buyAsset,
+          sendMax: false,
+          sellAmountBeforeFeesCryptoBaseUnit,
+          sellAsset,
+          receiveAddress: buyAssetReceiveAddr,
+          accountType: utxoAccountType || bitcoin.ChainAdapter.defaultUtxoAccountType,
+          accountNumber,
+          xpub: publicKey?.xpub || '',
+        })
+
+        maybeTrade.match<void | Promise<void>>({
+          ok: async trade => {
+            const tradeResult = await swapper.executeTrade({ trade, wallet })
+            console.info('broadcast tx with id: ', tradeResult.tradeId)
+          },
+          err: e => {
+            console.error(e)
+          },
+        })
+      }
+    },
```

Comment:
I know we're not refactoring here but this seems like it should have its own function

Reply:
Definitely, also worth asking ourselves if anyone has actually used swapper CLI over the last year before we put more effort into it

Reply:
I have not...

Reply:
Delete it?
![keyandpeeledeleteit](https://y.yarn.co/33b7d91a-af60-4ad8-84e8-5340260b21bd_thumb.jpg)

Reply:
I think we have consensus to kill it, but it won't block this PR 👌

Reply:
let's just delet while we're in here

Reply:
Deleted in aab97060e1fcf9f7f124222be501789583cd5a31, she will (not) be missed 👋🏽